### PR TITLE
[test_vlan] Test case for QinQ packet switched based on outer vlan tag

### PR DESCRIPTION
**Pull request by anish-n**
### Description of PR

Summary:
Vlan test enhancement to test L2 packet switching for QinQ packets. We test if the packet is switched based on the outer QinQ vlan tag, given tagged vlan configuration for associated port on the switch. 

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
To test packet switching for QinQ packets
#### How did you do it?
Enhance test_vlan to include PTF based packet test where we send QinQ packet
#### How did you verify/test it?
Tested on DUT where sonic-mgmt test_vlan was run
#### Any platform specific information?
Currently tested on the Mellanox platform: limited the test case for Mellanox only and marked pytest skip for others. Once it is tested on other platforms, the pytest skip list may be modified/platform check removed altogether.

### CODE REVIEW

**Review by kktheballer — APPROVED** [review]
Looks good to me and follows similar structure of the previous test cases

File: tests/vlan/test_vlan.py
Comment (theasianpianist) [inline tests/vlan/test_vlan.py:515]: Can we use `testutils.verify_packet` instead of `dp_poll`?

Comment (anish-n) [reply tests/vlan/test_vlan.py]: Thanks for the comment, changed to ```testutils.verify_packet```

**Review by theasianpianist — APPROVED** [review]
lgtm, please wait for review from @Azure/sonic-fundamentals

**Pull request merged**
